src/image.rs

```diff
@@ -133,6 +133,80 @@ impl ImageFormat {
         inner(path.as_ref())
     }
 
+    /// Return if the ImageFormat can be decoded by the lib.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use image::ImageFormat;
+    /// #[cfg(feature = "png")]
+    /// assert_eq!(ImageFormat::Png.can_read(), true);
+    /// ```
+    #[inline]
+    pub fn can_read(&self) -> bool {
+        // Needs to be updated once a new variant's decoder is added to free_functions.rs::load
+        match self {
+            #[cfg(feature = "png")]
+            ImageFormat::Png => true,
+            #[cfg(feature = "gif")]
+            ImageFormat::Gif => true,
+            #[cfg(feature = "jpeg")]
+            ImageFormat::Jpeg => true,
+            #[cfg(feature = "webp")]
+            ImageFormat::WebP => true,
+            #[cfg(feature = "tiff")]
+            ImageFormat::Tiff => true,
+            #[cfg(feature = "tga")]
+            ImageFormat::Tga => true,
+            #[cfg(feature = "dds")]
+            ImageFormat::Dds => true,
+            #[cfg(feature = "bmp")]
+            ImageFormat::Bmp => true,
+            #[cfg(feature = "ico")]
+            ImageFormat::Ico => true,
+            #[cfg(feature = "hdr")]
+            ImageFormat::Hdr => true,
+            #[cfg(feature = "pnm")]
+            ImageFormat::Pnm => true,
+            #[cfg(feature = "farbfeld")]
+            ImageFormat::Farbfeld => true,
+            _ => false,
```

Comment:
Could you replace this line with a match on the non-exhaustive variant? This will trigger a compile error if a new variant is added without this function being updated, which is what we want.
```rust
ImageFormat::__NonExhaustive(marker) => match marker._private {},
```